src/Console/ConsoleApplication.php

```diff
@@ -41,10 +41,10 @@ public function __construct(array $commands)
 
     public function doRun(InputInterface $input, OutputInterface $output): int
     {
-        // @fixes https://github.com/rectorphp/rector/issues/2205
         $isXdebugAllowed = $input->hasParameterOption('--xdebug');
-        if ($isXdebugAllowed) {
+        if (!$isXdebugAllowed) {
```

Comment:
I am curious why negation is needed?

Reply:
Calling `xDebugHandler->check()` will restart the process when xdebug is loaded. Therefore we should only call into this method when xdebug is not allowed

Reply:
Before this PR the worker command already contained the `--xdebug` flag. 

The code change here is related to how rector behaves when `--xdebug` is passed, which was inverted before this PR